statsmodels/tsa/stattools.py

```diff
@@ -1735,6 +1735,282 @@ def _sigma_est_kpss(resids, nobs, lags):
+            * 'c' : constant only (default)
+            * 't' : trend only
+            * 'ct' : constant and trend
+        autolag : {'AIC', 'BIC', 't-stat', None}
+            * if None, then maxlag lags are used
+            * if 'AIC' (default) or 'BIC', then the number of lags is chosen
+              to minimize the corresponding information criterion
+            * 't-stat' based choice of maxlag.  Starts with maxlag and drops a
+              lag until the t-statistic on the last lag length is significant
+              using a 5%-sized test
+
+        Returns
+        -------
+        zastat : float
+            test statistic
+        pvalue : float
+            based on MC-derived critical values
+        cvdict : dict
+            critical values for the test statistic at the 1%, 5%, and 10%
+            levels
+        bpidx : int
+            index of x corresponding to endogenously calculated break period
+            with values in the range [0..nobs-1]
+        baselag : int
+            number of lags used for period regressions
+
+        Notes
+        -----
+        H0 = unit root with a single structural break
+
+        Algorithm follows Baum (2004/2015) approximation to original
+        Zivot-Andrews method. Rather than performing an autolag regression at
+        each candidate break period (as per the original paper), a single
+        autolag regression is run up-front on the base model (constant + trend
+        with no dummies) to determine the best lag length. This lag length is
+        then used for all subsequent break-period regressions. This results in
+        significant run time reduction but also slightly more pessimistic test
+        statistics than the original Zivot-Andrews method, although no attempt
+        has been made to characterize the size/power tradeoff.
+
+        References
+        ----------
+        Baum, C.F. (2004). ZANDREWS: Stata module to calculate Zivot-Andrews
+        unit root test in presence of structural break," Statistical Software
+        Components S437301, Boston College Department of Economics, revised
+        2015.
+
+        Schwert, G.W. (1989). Tests for unit roots: A Monte Carlo
+        investigation. Journal of Business & Economic Statistics, 7: 147-159.
+
+        Zivot, E., and Andrews, D.W.K. (1992). Further evidence on the great
+        crash, the oil-price shock, and the unit-root hypothesis. Journal of
+        Business & Economic Studies, 10: 251-270.
+        """
+        if regression not in ['c', 't', 'ct']:
+            raise ValueError(
+                'ZA: regression option \'%s\' not understood' % regression)
+        if not isinstance(trim, float) or trim < 0 or trim > (1. / 3.):
+            raise ValueError(
+                'ZA: trim value must be a float in range [0, 0.333]')
+        x = np.asarray(x)
+        if x.ndim > 2 or (x.ndim == 2 and x.shape[1] != 1):
+            raise ValueError(
```

Comment:
quick tests for the failure modes?

Reply:
Added failure mode tests to test_stattools.py.